### ISSUE TITLE
Lint unnecessary transmutes from ref to ptr

### ISSUE THREAD

**Issue by crumblingstatue** [Manishearth/rust-clippy#930]
We already have `transmute_ptr_to_ref`, but we could also lint the opposite case.

```
 fn type_(evt: &mut sfEvent) -> *mut sfEventType {
 -    unsafe { ::std::mem::transmute(evt) }
 +    evt as *mut _ as _
  }
```

```
 fn size(evt: &mut sfEvent) -> event::Event {
 -    let e: *mut sfSizeEvent = unsafe { ::std::mem::transmute(evt) };
 +    let e: *mut sfSizeEvent = evt as *mut _ as _;
      unsafe { event::Resized { width: (*e).width, height: (*e).height } }
  }
```

Ref https://github.com/jeremyletang/rust-sfml/commit/d84ce7543ed8af55c8c85c346c72c3204e7a1890?diff=unified

Edit: The `as _` trick for the second `as` only works on 1.10 nightly (https://github.com/rust-lang/rust/issues/33657)

**Comment by oli-obk** [comment]
fixed by #1045

**Comment by llogiq** [comment]
Another one bites the dust... :smile:
